SourceX/controls/keymapper.hpp

```diff
@@ -0,0 +1,77 @@
+#pragma once
```

Comment:
1. The header extension we use throughout the codebase is `.h`, `.hpp`.
2. Please make sure the new files use CRLF line endings (simply resaving in VS code should do it now that you have the extension installed).

Reply:
>     1. The header extension we use throughout the codebase is `.h`, `.hpp`.

I had a discussion about that on Discord and @AJenbo seemed to be fine with migrating to `.hpp`, which makes more sense since those are C++ headers, not C ones.

>     2. Please make sure the new files use CRLF line endings (simply resaving in VS code should do it now that you have the extension installed).

Yes that should be the case now.